pandas/tests/frame/test_arithmetic.py

```diff
@@ -931,16 +931,16 @@ def test_binop_other(self, op, value, dtype):
         elif (op, dtype) in skip:
 
             if op in [operator.add, operator.mul]:
-                with tm.assert_produces_warning(UserWarning):
-                    # "evaluating in Python space because ..."
-                    op(s, e.value)
+                # TODO we should assert this or not depending on whether
+                # numexpr is used or not
+                # with tm.assert_produces_warning(UserWarning):
+                #     # "evaluating in Python space because ..."
+                op(s, e.value)
 
             else:
                 msg = "operator '.*' not implemented for .* dtypes"
                 with pytest.raises(NotImplementedError, match=msg):
-                    with tm.assert_produces_warning(UserWarning):
-                        # "evaluating in Python space because ..."
-                        op(s, e.value)
+                    op(s, e.value)
```

Comment:
This was a wrong test, but because putting `assert_produces_warning` inside `pytest.raises` doesn't fail if it doesn't warn, this wasn't noticed. 
The ops for which we raise a warning vs raise NotImplementedError are not overlapping.